discovery_engine/lib/src/generators/multiplier_generator.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+import 'package:source_gen/source_gen.dart';
+
+import 'annotations.dart';
+import 'class_visitor.dart';
+
+class MultiplierGenerator extends GeneratorForAnnotation<MapEvent> implements ElementVisitor<Element> {
```

Comment:
can this implementation be simplified such that we only override those methods which currently do more than a print statement?

it also looks like the [lib authors](https://pub.dev/documentation/analyzer/latest/dart_element_element/ElementVisitor-class.html) didn't intent to implement `ElementVisitor` directly, but one of its subclasses, but i guess we could easily rewrite this to eg implement `SimpleElementVisitor` with some overrides instead.

Reply:
Yep, I just didn't want to invest anymore time than necessary until I know we're going to use this. I would employ a similar approach as for the `class_visitor`, but I didn't know about that yet when I created this class :she